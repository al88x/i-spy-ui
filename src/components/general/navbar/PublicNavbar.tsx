import React, { useContext } from 'react';
import { Link, Redirect } from "react-router-dom";
import './PublicNavbar.scss'
import { AuthContext } from '../../AuthContext';

export function PublicNavbar() {
    const context = useContext(AuthContext);

    function handleClick() {
        context.setToken("");
    }
    
    if(context.loggedIn) {
        return (
            <nav>
                <div className="main-nav-elements">
                    <div>
                        <Link to="/" className="nav-element">Home</Link>
                    </div>
                    <div>
                        <Link to="/reports" className="nav-element">Reports</Link>
                    </div>
                    <div className="super-admin">
                        <Link to="/users" className="nav-element">Manage Users</Link>
                    </div>
                    <div className="super-admin">
                        <Link to="/settings" className="nav-element">Admin Settings</Link>
                    </div>
                </div>
                <div className="logOut" >
                    <Link to="/" className="nav-element" onClick={handleClick}>Log Out</Link>
                </div>
            </nav>
        );
    }

    return (
            <nav>
                <div className="main-nav-elements">
                    <div>
                        <Link to="/" className="nav-element">Home</Link>
                    </div>
                </div>
                <div className="login">
                    <div>
                        <Link to="/login" className="nav-element">Login</Link>
                    </div>
                </div>
            </nav>
    );
}

