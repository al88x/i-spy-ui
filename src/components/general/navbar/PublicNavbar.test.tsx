import React from 'react';
import { PublicNavbar } from './PublicNavbar';
import renderer from 'react-test-renderer';
import { BrowserRouter} from "react-router-dom";

describe("Navbar", () => {
    it("Should render without errors", () => {
        const component = renderer.create(
            <BrowserRouter>
            <PublicNavbar/>
            </BrowserRouter>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
