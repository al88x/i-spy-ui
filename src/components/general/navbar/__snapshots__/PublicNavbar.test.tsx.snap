// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render without errors 1`] = `
<nav>
  <div
    className="main-nav-elements"
  >
    <div>
      <a
        className="nav-element"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </div>
  <div
    className="login"
  >
    <div>
      <a
        className="nav-element"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </div>
</nav>
`;
